src/display/annotation_layer.js

```diff
@@ -1007,8 +1008,15 @@ var FileAttachmentAnnotationElement = (
   function FileAttachmentAnnotationElement(parameters) {
     AnnotationElement.call(this, parameters, true);
 
-    this.filename = getFilenameFromUrl(parameters.data.file.filename);
-    this.content = parameters.data.file.content;
+    var file = this.data.file;
+    this.filename = getFilenameFromUrl(file.filename);
+    this.content = file.content;
+
+    this.linkService.onFileAttachmentAnnotation({
+      id: stringToPDFString(file.filename),
```

Comment:
> I'm a bit confused why we don't use this.filename and this.content in these three lines instead.

I wanted to make sure that what we (eventually) pass on to `pdf_attachment_viewer.js` has the *exact* same format as what `PDFDocumentProxy_getAttachments` returns, by explicitly using the file attachment data from the API side here, to avoid any possible issues/surprises later on.

> Especially the file name has been sanitized then.

Note also that we already need to use `getFilenameFromUrl` in [`pdf_attachment_viewer.js`](https://github.com/mozilla/pdf.js/blob/a917443ae6733e77ec0011097ddfb23fa0e740fc/web/pdf_attachment_viewer.js#L112), so using it here as well does not help us much and again wouldn't be compatible with what we get from `PDFDocumentProxy_getAttachments`.